src/lib/formatters/test/format-test-results.ts

```diff
@@ -43,16 +43,23 @@ function createJsonResultOutput(jsonResult, options: Options) {
   return jsonResultClone;
 }
 
-function formatJsonVulnerabilityStructure(jsonResult, options: Options) {
+export function formatJsonVulnerabilityStructure(jsonResult, options: Options) {
   if (options['group-issues']) {
     jsonResult.vulnerabilities = Object.values(
       (jsonResult.vulnerabilities || []).reduce((acc, vuln): Record<
         string,
         any
       > => {
-        vuln.from = [vuln.from].concat(acc[vuln.id]?.from || []);
-        vuln.name = [vuln.name].concat(acc[vuln.id]?.name || []);
-        acc[vuln.id] = vuln;
+        if (!acc[vuln.id]) {
+          // initialize new vuln.id with a 'from' and 'name' arrays
+          acc[vuln.id] = vuln;
+          acc[vuln.id].from = [vuln.from];
+          acc[vuln.id].name = [vuln.name];
```

Comment:
Not a fan of mutating the original vuln. Consider:
```suggestion
          acc[vuln.id] = {
            ...vuln,
            from: [vuln.from],
            name: [vuln.name],
          };
```